package/src/main.rs

```diff
@@ -1127,30 +1123,43 @@ index b70ae75..271706a 100644
         test_pants_from_sources(
             Command::new(pants_from_sources)
                 .env("SCIE_PANTS_TEST_MODE", "pants_from_sources mode")
+                .env("PANTS_VENV_DIR_PREFIX", &pants_2_14_1_venv_dir)
                 .current_dir(user_repo_dir),
-            "The pants_from_sources mode is working.",
+            vec![
+                "The pants_from_sources mode is working.",
+                "Pants from sources argv: --no-verify-config -V.",
+            ],
         )?;
 
         integration_test!("Verify delegating to `./pants`.");
-        let result = execute(
+        test_pants_from_sources(
             Command::new(scie_pants_scie)
-                .arg("-V")
                 .env("SCIE_PANTS_TEST_MODE", "delegate_bootstrap mode")
-                .current_dir(pants_2_14_1_clone_dir)
-                .stderr(Stdio::piped()),
+                .current_dir(&pants_2_14_1_clone_dir),
+            vec![
+                "The delegate_bootstrap mode is working.",
+                "Pants from sources argv: -V.",
+            ],
         )?;
-        let stderr = String::from_utf8(result.stderr).map_err(|e| {
-            Code::FAILURE.with_message(format!("Failed to decode Pants stderr: {e}"))
-        })?;
-        let expected_message = "The delegate_bootstrap mode is working.";
-        assert!(
-            stderr.contains(expected_message),
-            "STDERR did not contain '{expected_message}':\n{stderr}"
-        );
-        let expected_argv_message = "Pants from sources argv: -V.";
+
+        integration_test!("Verify usage of a released Pants version on the pants repo.");
+        let stderr = test_pants_from_sources(
+            Command::new(scie_pants_scie)
+                .env("PANTS_VERSION", "2.16.0.dev5")
+                .env(
+                    "PANTS_BACKEND_PACKAGES",
+                    "-[\
+                    'internal_plugins.test_lockfile_fixtures',\
+                    'pants.backend.explorer',\
+                    ]",
+                )
+                .current_dir(&pants_2_14_1_clone_dir),
+            vec!["[INFO] Scheduler initialized."],
```

Comment:
Yeah, this could be flaky maybe? It relies on the conditions dictating pantsd must restart. Here the underlying codebase is identical and just the ~PANTS_VERSION env var has changed which may not be enough to change the pantsd fingerprint which is the only thing guaranteed to cause a pantsd restart. The rest are due to exceeding memory thresholds.

I'm definitely a bit muddy on the above, but an alternative is to do a more direct test. Have the Command capture stdout and re-jigger the test closure so you can get at stdout / the process result and just verify the printed version is 2.16.0.dev5. Other ITs do this sort of thing.

Reply:
Right here for example, although with stderr:
https://github.com/pantsbuild/scie-pants/blob/7fc1471d6d0d66615fb0217459d5942cd161a799/package/src/main.rs#L819-L833

Reply:
thanks, will circle back to that later :)